pkg/storage/migrations/migration.go

```diff
@@ -399,11 +445,157 @@ func convertOutput(result storage.Result, outputKey string, data []byte) (storag
 }
 
 func (m *Migration) migrateCredentialSets(ctx context.Context) error {
-	panic("not implemented")
+	ctx, span := tracing.StartSpan(ctx)
+	defer span.EndSpan()
+
+	// Get a list of all the credential set names
+	names, err := m.listItems("credentials", "")
+	if err != nil {
+		return span.Error(fmt.Errorf("error listing credential sets from the source account: %w", err))
+	}
+
+	span.Infof("Found %d credential sets to migrate", len(names))
+
+	var bigErr *multierror.Error
+	for _, name := range names {
+		if err = m.migrateCredentialSet(ctx, name); err != nil {
+			// Keep track of which ones failed but otherwise keep trying to migrate as many as possible
+			bigErr = multierror.Append(bigErr, err)
+		}
+	}
+
+	return bigErr.ErrorOrNil()
+}
+
+func (m *Migration) migrateCredentialSet(ctx context.Context, name string) error {
+	ctx, span := tracing.StartSpan(ctx, attribute.String("credential-set", name))
+	defer span.EndSpan()
+
+	data, err := m.sourceStore.Read("credentials", name)
+	if err != nil {
+		return span.Error(err)
+	}
+
+	dest, err := convertCredentialSet(m.opts.NewNamespace, data)
+	if err != nil {
+		return span.Error(err)
+	}
+
+	err = m.destStore.Insert(ctx, storage.CollectionCredentials, storage.InsertOptions{Documents: []interface{}{dest}})
```

Comment:
Would this result in an error if some old record has been migrated? If so, should we just ignore duplicated key error?

Reply:
I'll switch to upsert to avoid the problem.